components/raftstore/src/store/fsm/store.rs

```diff
@@ -660,8 +664,27 @@ impl<EK: KvEngine, ER: RaftEngine, T: Transport> RaftPoller<EK, ER, T> {
         {
             self.poll_ctx.trans.flush();
         }
-        let ready_cnt = self.poll_ctx.ready_res.len();
+        let ready_cnt = self.poll_ctx.ready_res.len() + self.poll_ctx.readonly_ready_res.len();
+        if !self.poll_ctx.readonly_ready_res.is_empty() {
+            let mut readonly_ready_res = mem::take(&mut self.poll_ctx.readonly_ready_res);
+            for ready in readonly_ready_res.drain(..) {
+                to_skip_end.push(ready.batch_offset);
+                PeerFsmDelegate::new(
+                    peers[ready.batch_offset].as_mut().unwrap(),
+                    &mut self.poll_ctx,
+                )
+                .post_raft_ready_append(ready);
+            }
+            self.poll_ctx.readonly_ready_res = readonly_ready_res;
```

Comment:
```suggestion
```
`mem::take` will replace `self.poll_ctx.readonly_ready_res` with `Vec::default`

Reply:
Yes, I reassign to avoid future allocations.